File: src/main/java/com/redbox/domain/funding/repository/FundingRepositoryCustom.java
package com.redbox.domain.funding.repository;

import com.redbox.domain.funding.dto.FundingFilter;
import com.redbox.domain.funding.entity.Funding;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

public interface FundingRepositoryCustom {
    public Page<Funding> searchBoards(Long userId, FundingFilter fundingFilter, Pageable pageable);
}

File: migration/src/main/kotlin/com/redbox/domain/community/funding/repository/FundingRepositoryCustom.kt
package com.redbox.domain.funding.repository

import com.redbox.domain.community.funding.dto.FundingFilter
import com.redbox.domain.community.funding.entity.Funding
import org.springframework.data.domain.Page
import org.springframework.data.domain.Pageable

interface FundingRepositoryCustom {
    fun searchBoards(userId: Long?, fundingFilter: FundingFilter, pageable: Pageable): Page<Funding>
}
